### PR TITLE
useFactory shouldn't accept connectionName as it's part of MongooseModuleAsyncOptions

### PR DESCRIPTION
**Pull request by ledniy**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently, you can pass connectionName both inside forRootAsync as well as inner useFactory, although only connectionName from the root of forRootAsync will be used

Issue Number: N/A

## What is the new behavior?
Only forRootAsync will accept connectionName

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: lib/interfaces/mongoose-options.interface.ts [hunk 1]
```diff
@@ -18,13 +18,15 @@ export interface MongooseOptionsFactory {
     | MongooseModuleOptions;
 }
 
+export type MongooseModuleOptionsFactory = Omit<MongooseModuleOptions, 'connectionName'>
```
Comment (kamilmysliwiec) [inline lib/interfaces/mongoose-options.interface.ts:21]: The name here is somewhat confusing as this isn't a factory but rather options object that must be passed **into** a factory function.

Comment (ledniy) [reply lib/interfaces/mongoose-options.interface.ts:21]: any idea for better naming? `MongooseModuleFactoryOptions`?

Comment (kamilmysliwiec) [reply lib/interfaces/mongoose-options.interface.ts:21]: sounds good

Comment (ledniy) [reply lib/interfaces/mongoose-options.interface.ts]: done

**Review by Tony133 — APPROVED** [review]

**Pull request merged**